server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
Same would than be true for ForwardingDownstreamAdapter or all other places, where we want to use the metrics. The question is then for me if it would not be better to null check on each metrics call itself, because metrics are an aspect and not needed for base functionality. I think the metrics would not be used outside of a spring managed environment.